src/api/BitbucketVersionControl.ts

```diff
@@ -61,52 +61,234 @@ export default class BitbucketVersionControl implements VersionControlAbstractio
+				// file is not yet created; remove commit ID
+				lastCommitId = '';
+			}
+
+			const logToPush = ChangelogFormatter.insertNewEntryIntoLog(currentChangelog, newContent);
+
+			await this.pushNewChangelog(logToPush, commitMessage, lastCommitId);
+			this.loggerReference.logMessage(LogLevels.info, 'BitbucketVersionControl: new changelog pushed successful');
+		} catch (error: any) {
+			this.loggerReference.logMessage(LogLevels.error, error.message);
+			return false;
+		}
+		// signal that everything worked
+		return true;
+	}
+
+	/**
+	 * Internal function to get the list of branches in the repository
+	 * @returns An object that contains the base branch name as well as an array with all branches in the repository
+	 */
+	private async getBranchList(): Promise<{ baseBranch: string; availableBranches: Array<string>; }> {
+		const result = {
+			baseBranch: '',
+			availableBranches: new Array<string>()
+		}
+		const bitbucketApiAuthHeader = {
+			auth: {
+				username: this.authentication.user,
+				password: this.authentication.password
+			}
+		};
+		this.loggerReference.logMessage(LogLevels.debug, 'BitbucketVersionControl: fetching list of branches in repository');
+		let response: any = undefined;
+		try {
+			const requestResponse = await axios.get(
+				`${this.serverConfig.serverUrl}/rest/api/1.0/projects/${this.serverConfig.projectName}/repos/${this.serverConfig.repositoryName}/branches?limit=1000`,
+				bitbucketApiAuthHeader
+			);
+			response = requestResponse;
+		} catch (error: any) {
+			this.loggerReference.logObject(LogLevels.debug, 'Axios error result:', error.response.data.errors);
+		}
+		if (response.status !== 200) {
+			throw this.exceptionProvider.getNewException(ExceptionTypes.UnexpectedResponseException,
+					'BitbucketVersionControl: could not retrieve list of branches');
+		}
+		if (!response.data.isLastPage) {
+			throw this.exceptionProvider.getNewException(ExceptionTypes.UnexpectedResponseException,
+					'BitbucketVersionControl: found more then 1000 branches in the repository, canceling changelog commit - please clean up to reactivate changelog writing');
+		}
+		response.data.values.forEach((branch: any) => {
+			if (branch.isDefault) {
+				result.baseBranch = branch.displayId;
+			}
+			result.availableBranches.push(branch.displayId as string);
+		});
+		return result;
+	}
+
+	/**
+	 * Internal function to create a branch in the repository
+	 * @param branchName The name of the branch that should be created
+	 * @param startCommit The ID of the commit on the base-branch that should be used as origin
+	 */
+	private async createBranch(branchName: string, startCommit: string): Promise<void> {
+		const bitbucketApiAuthHeader = {
+			auth: {
+				username: this.authentication.user,
+				password: this.authentication.password
+			}
+		};
+		this.loggerReference.logMessage(LogLevels.debug, 'BitbucketVersionControl: creating working branch in repository');
+		let responseStatus = undefined;
```

Comment:
in previous method this was given type `any`

Reply:
good find, changed